classes/ratings_and_allocations_table.php

```diff
@@ -366,11 +374,13 @@ private function add_summary_row() {
         }
 
         foreach ($this->choicesum as $choiceid => $sum) {
-            $row[] = get_string(
+            if (array_key_exists($choiceid, $this->filter_choiceids(array_keys($this->choicenames)))) {
```

Comment:
When no groupfilter is selected, the keys of the array are empty. Better check with `in_array()`